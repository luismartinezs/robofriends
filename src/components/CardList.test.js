import React from "react";
import { shallow, mount, render } from "enzyme";
// most of the time use shallow
// mount: full DOM render, requires full DOM API (JS DOM, headless browser, actual browser...)
// render: render components to static HTML (using cheerio lib), renders any children of current component (unlike shallow)
import CardList from "./CardList";

it("expect to render Card component", () => {
  const mockRobots = [{
		id: 1,
		name: "Jon Snow",
		username: "JonJon",
		email: "jon@winterfell.com"
	}]
  expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
});
