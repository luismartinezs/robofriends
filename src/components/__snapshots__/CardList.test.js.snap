// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList component 1`] = `
<div>
  <Card
    email="jon@winterfell.com"
    id={1}
    key="1"
    name="Jon Snow"
  />
</div>
`;
